[None][chore] cosmetic changes to the configs database.

fadi-sadi opened the pull request:
- This change is purely cosmetic.
- Sorting is done based on the order of the fields in the first entry.
- Added a pre-commit hook to ensure the file is sorted on every update.
- Aligned gpu name in recipe file paths with lookup data.
- Moved all config database scripts to scripts/config_database.
- Updated config database tests.

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **Chores**
  * Added automated sorting for configuration files via pre-commit hook to maintain consistent organization

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

## Description

These changes sort the database lookup.yaml file, to make tracking changes easier and to avoid having recipes of the same model scattered in different parts of the lookup file when manually going over the entries.
The plan is to start adding recipe updates frequently and it is easier to understand what gaps we are filling when the data is sorted.
A pre-commit hook was added to ensure the file stays sorted after every change.

## Test Coverage


## PR Checklist

Please review the following before submitting your PR:
- PR description clearly explains what and why. If using CodeRabbit's summary, please make sure it makes sense.
- PR Follows [TRT-LLM CODING GUIDELINES](https://github.com/NVIDIA/TensorRT-LLM/blob/main/CODING_GUIDELINES.md) to the best of your knowledge.
- Test cases are provided for new code paths (see [test instructions](https://github.com/NVIDIA/TensorRT-LLM/tree/main/tests#1-how-does-the-ci-work))
- Any new dependencies have been scanned for license and vulnerabilities
- [CODEOWNERS](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/CODEOWNERS) updated if ownership changes
- Documentation updated as needed
- Update [tava architecture diagram](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/tava_architecture_diagram.md) if there is a significant design change in PR.
- The reviewers assigned automatically/manually are appropriate for the PR.


- [x] Please check this after reviewing the above items as appropriate for this PR.

## GitHub Bot Help

`/bot [-h] ['run', 'kill', 'skip', 'reuse-pipeline'] ...`

Provide a user friendly way for developers to interact with a Jenkins server.

Run `/bot [-h|--help]` to print this help message.

See details below for each supported subcommand.

<details>

`run  [--reuse-test (optional)pipeline-id --disable-fail-fast --skip-test --stage-list "A10-PyTorch-1, xxx" --gpu-type "A30, H100_PCIe" --test-backend "pytorch, cpp" --add-multi-gpu-test --only-multi-gpu-test --disable-multi-gpu-test --post-merge --extra-stage "H100_PCIe-TensorRT-Post-Merge-1, xxx" --detailed-log --debug(experimental)]`

Launch build/test pipelines. All previously running jobs will be killed.

`--reuse-test (optional)pipeline-id ` *(OPTIONAL)* : Allow the new pipeline to reuse build artifacts and skip successful test stages from a specified pipeline or the last pipeline if no pipeline-id is indicated. If the Git commit ID has changed, this option will be always ignored. The DEFAULT behavior of the bot is to reuse build artifacts and successful test results from the last pipeline.

`--disable-reuse-test ` *(OPTIONAL)* : Explicitly prevent the pipeline from reusing build artifacts and skipping successful test stages from a previous pipeline. Ensure that all builds and tests are run regardless of previous successes.

`--disable-fail-fast ` *(OPTIONAL)* : Disable fail fast on build/tests/infra failures.

`--skip-test ` *(OPTIONAL)* : Skip all test stages, but still run build stages, package stages and sanity check stages. Note: Does **NOT** update GitHub check status.

`--stage-list "A10-PyTorch-1, xxx"` *(OPTIONAL)* : Only run the specified test stages. Examples: "A10-PyTorch-1, xxx". Note: Does **NOT** update GitHub check status.

`--gpu-type "A30, H100_PCIe"` *(OPTIONAL)* : Only run the test stages on the specified GPU types. Examples: "A30, H100_PCIe". Note: Does **NOT** update GitHub check status.

`--test-backend "pytorch, cpp"` *(OPTIONAL)* : Skip test stages which don't match the specified backends. Only support [pytorch, cpp, tensorrt, triton]. Examples: "pytorch, cpp" (does not run test stages with tensorrt or triton backend). Note: Does **NOT** update GitHub pipeline status.

`--only-multi-gpu-test ` *(OPTIONAL)* : Only run the multi-GPU tests. Note: Does **NOT** update GitHub check status.

`--disable-multi-gpu-test ` *(OPTIONAL)* : Disable the multi-GPU tests. Note: Does **NOT** update GitHub check status.

`--add-multi-gpu-test ` *(OPTIONAL)* : Force run the multi-GPU tests in addition to running L0 pre-merge pipeline.

`--post-merge ` *(OPTIONAL)* : Run the L0 post-merge pipeline instead of the ordinary L0 pre-merge pipeline.

`--extra-stage "H100_PCIe-TensorRT-Post-Merge-1, xxx"` *(OPTIONAL)* : Run the ordinary L0 pre-merge pipeline and specified test stages. Examples: --extra-stage "H100_PCIe-TensorRT-Post-Merge-1, xxx".

`--detailed-log ` *(OPTIONAL)* : Enable flushing out all logs to the Jenkins console. This will significantly increase the log volume and may slow down the job.

`--debug ` *(OPTIONAL)* : **Experimental feature**. Enable access to the CI container for debugging purpose. Note: Specify exactly one stage in the `stage-list` parameter to access the appropriate container environment. Note: Does **NOT** update GitHub check status.

For guidance on mapping tests to stage names, see `docs/source/reference/ci-overview.md`
and the `scripts/test_to_stage_mapping.py` helper.

### kill

`kill  `

Kill all running builds associated with pull request.

### skip

`skip --comment COMMENT `

Skip testing for latest commit on pull request. `--comment "Reason for skipping build/test"` is required. IMPORTANT NOTE: This is dangerous since lack of user care and validation can cause top of tree to break.

### reuse-pipeline

`reuse-pipeline `

Reuse a previous pipeline to validate current commit. This action will also kill all currently running builds associated with the pull request. IMPORTANT NOTE: This is dangerous since lack of user care and validation can cause top of tree to break.

</details>

fsaady committed:
[None][chore] sort database lookup.yaml for easier updates.

- Sorting is done based on the order of the fields in the first entry.
- Added a pre-commit hook to ensure the file is sorted on every update.

Signed-off-by: Fadi Saady <fsaady@nvidia.com>

fadi-sadi commented on scripts/sort_lookup_yaml.py:
```diff
@@ -0,0 +1,43 @@
+import argparse
+import sys
+
+import yaml
```
fixed

anish-shanbhag commented on scripts/sort_lookup_yaml.py:
```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/env python3
```
Maybe we should move the 3 config DB scripts (`scripts/generate_config_database_tests.py`, `scripts/generate_config_table.py`, and this one) either to a subdir of `scripts` or into `examples/configs/scripts`

fadi-sadi commented on .pre-commit-config.yaml:
```diff
@@ -1472,3 +1472,9 @@ repos:
         entry: ./scripts/dco_check.py
         language: script
         stages: [commit-msg]
+    -   id: sort-lookup-yaml
```
I like the idea of combining, but I'm not completely sure they should all be pre-commit hooks. The way I see it, a hook should be able to modify a failed file in a way that would make it pass when ran again. With the generation of the config tables, the data might be correct but a script issue might cause the failure, in which case the modification is not very straight forward, which is why I think it's more suitable for a unit test. Does it make sense or am I missing something?

fadi-sadi committed:
Merge branch 'main' into fsaady/sort-lookup

fadi-sadi committed:
Merge branch 'main' into fsaady/sort-lookup

fadi-sadi committed:
Merge branch 'NVIDIA:main' into fsaady/sort-lookup

fsaady committed:
chore(configs/database): cosmetic changes to configs database.

- aligning gpu name in recipe file paths with lookup data.
- moving all config database scripts to scripts/config_database.
- updating config database tests.

Signed-off-by: Fadi Saady <fsaady@nvidia.com>

fadi-sadi committed:
Merge branch 'NVIDIA:main' into fsaady/sort-lookup

fsaady committed:
chore(configs/database): fixing config scripts to use in pre-commit.

Signed-off-by: Fadi Saady <fsaady@nvidia.com>

fadi-sadi committed:
Merge branch 'main' into fsaady/sort-lookup

fadi-sadi committed:
Merge branch 'main' into fsaady/sort-lookup

tensorrt-cicd commented:
[PR_Github #33916](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/33916/) [ run ] triggered by Bot. Commit: `27efb00`

tensorrt-cicd commented:
[PR_Github #33916](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/33916/) [ run ] completed with state `SUCCESS`. Commit: `27efb00` 
[/LLM/main/L0_MergeRequest_PR pipeline #26156](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/26156/) completed with status: 'FAILURE'

⚠️ **Action Required:**
- Please check the failed tests and fix your PR
- If you cannot view the failures, ask the CI triggerer to share details
- Once fixed, request an NVIDIA team member to trigger CI again

fsaady committed:
chore(configs/database): fixing ruff-format pre-commit failure.

Signed-off-by: Fadi Saady <fsaady@nvidia.com>

fsaady commented:
Closing this PR for now.
Will submit 2 updated PRs for the gpu name alignment and for sorting the lookup file.
